### PR TITLE
Add Alcide Kubernetes Advisor

### PR DESCRIPTION
**Pull request by gadinaor**
Alcide Advisor is an agentless Kubernetes audit, compliance and hygiene scanner that’s built to ensure a friciton free DevSecOps workflows. Alcide Advisor can be plugged early in the development process and before moving to production.

### CODE REVIEW

**Comment by gadinaor** [comment]
@rphillips @matti @dlackty what's the merge cycle on those pull requests?

**Comment by matti** [comment]
long.

**Comment by sanastos** [comment]
Sorry, we haven't had dedicated coverage for reviewing these requests in a long time. We now have a rotation for this and things should be more prompt, although we have quite a backlog to work through.

I will take a look at this request shortly.

**Comment by sanastos** [comment]
Sorry for the delay, it took some time to get confirmation that we were allowed to merge this request.

After looking at this commit, it seems a bit confused to me. The cloudbuild.yaml file in the base directory should take care of building the image to be used, then you should have an examples directory with a cloudbuild.yaml that references the built image and executes the build step to do the scan.

Basically, you should make 3 changes:

1. Anyone (including me) should be able to do 'gcloud builds submit .' to build your builder. Strip out the unnecessary stuff.
1. There should also be a basic cloudbuild.yaml example file in the examples directory that just runs as a sanity check (you can execute --version or --help, or if there is anything more relevant).
1. There can additionally be cloudbuild-<usecase>.yaml files in the example directory that might need more input from the user to function (e.g. details about private keys). This is currently the latter part of your cloudbuild.yaml file.

**Comment by gadinaor** [comment]
@sanastos - updated PR - would appreciate if you can review

File: alcide/cloudbuild.yaml [hunk 1]
```diff
@@ -1,50 +1,13 @@
...
+# $ gcloud builds submit . --config=cloudbuild.yaml
+
 steps:
 #
 # Build the Alcide Advisor Image
 #
 - name: gcr.io/cloud-builders/docker
   args: ['build', '--tag=gcr.io/$PROJECT_ID/advisor', '.']
+- name: "gcr.io/${PROJECT_ID}/advisor"
+  args: ["validate", "cluster", "--help"]
```
Comment (sanastos) [inline alcide/cloudbuild.yaml]: No need to include this since you have a sanity check in examples/cloudbuild.yaml

File: alcide/cloudbuild.yaml [hunk 2]
```diff
@@ -1,50 +1,13 @@
...
-  - 'CLOUDSDK_CONTAINER_CLUSTER=${_GKE_CLUSTER}'
-  #- 'ALCIDE_APIKEY=${_ALCIDE_APIKEY}'
-  #- 'ALCIDE_API_SERVER=${_ALCIDE_API_SERVER}'
-
-# artifacts:
-#   objects:
-#     location: 'gs://[STORAGE_LOCATION]/'
-#     paths: ['scan.html']
\ No newline at end of file
+images: ["gcr.io/${PROJECT_ID}/advisor"]
```
Comment (sanastos) [inline alcide/cloudbuild.yaml]: Would you mind adding a newline to the end of the file here?

File: alcide/example/cloudbuild.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+steps:
+
+#
+# Test that Alcide Kubernetes Advisor builder exist in your project
+#
+- name: "gcr.io/${PROJECT_ID}/advisor"
+  args: ["validate", "cluster", "--help"]
+
```
Comment (sanastos) [inline alcide/example/cloudbuild.yaml]: There's an excess line here.

File: alcide/example/cloudbuild-gkescan.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  env:
+  - 'CLOUDSDK_COMPUTE_ZONE=${_GKE_CLUSTER_ZONE}'
+  - 'CLOUDSDK_CONTAINER_CLUSTER=${_GKE_CLUSTER}'
+  #- 'ALCIDE_APIKEY=${_ALCIDE_APIKEY}'
+  #- 'ALCIDE_API_SERVER=${_ALCIDE_API_SERVER}'
+
+# artifacts:
+#   objects:
+#     location: 'gs://[STORAGE_LOCATION]/'
+#     paths: ['scan.html']
```
Comment (sanastos) [inline alcide/example/cloudbuild-gkescan.yaml]: Newline here too.

**Comment by sanastos** [comment]
Thanks for your contribution!

**Pull request merged**